zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/FileInfo.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zeppelin.notebook;
+
+import java.util.concurrent.ConcurrentMap;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.collect.Maps;
+
+/**
+ * Represents note file info for storage 
+ *
+ */
+public class FileInfo {
+
+  private ConcurrentMap<String, String> fileName;
+  private boolean forceRename;
+  public static final FileInfo EMPTY = new FileInfo();
+  
+  private FileInfo() {
+    fileName = Maps.newConcurrentMap();
+    forceRename = false;
+  }
+  
+  public static FileInfo createInstance() {
+    return new FileInfo();
+  }
+  
+  public String getFileName(String storageName) {
+    if (!fileName.containsKey(storageName)) {
```

Comment:
right first time when renaming old file name is note.json, but here's couple of reasons to keep current filename:
 *  if user changes title of notebook further, it needs to be renamed again to reflect uptodate status.
 * having the old filename would help to handle the case of multiple notes under same folder
 * it would further help linking folder structure shown in Zeppelin with physical folder structure under notebooks folder